ebean-core/src/main/java/io/ebeaninternal/api/LoadBeanRequest.java

```diff
@@ -100,16 +90,21 @@ public int getBatchSize() {
    * Configure the query for lazy loading execution.
    */
   public void configureQuery(SpiQuery<?> query, List<Object> idList) {
-    query.setMode(SpiQuery.Mode.LAZYLOAD_BEAN);
+    query.setMode(Mode.LAZYLOAD_BEAN);
     query.setPersistenceContext(loadBuffer.getPersistenceContext());
-    String mode = isLazy() ? "+lazy" : "+query";
-    query.setLoadDescription(mode, getDescription());
-
-    if (isLazy()) {
-      // cascade the batch size (if set) for further lazy loading
+    query.setLoadDescription(lazy ? "+lazy" : "+query", getDescription());
+    if (lazy) {
       query.setLazyLoadBatchSize(getBatchSize());
+      if (alreadyLoaded) {
+        query.setBeanCacheMode(CacheMode.OFF);
```

Comment:
Explicitly NOT using the bean cache when lazy loading a bean that is already loaded (a partial)